nuxeo-routing-core/src/main/java/org/nuxeo/ecm/platform/routing/core/impl/DocumentRoutingServiceImpl.java

```diff
@@ -981,35 +988,22 @@ private WfCleaner(String repositoryName, int limit) {
 
         @Override
         public void run() {
-            List<String> routeIds = new ArrayList<>();
-            String query = "SELECT ecm:uuid FROM DocumentRoute WHERE (ecm:currentLifeCycleState = 'done' "
-                    + "OR ecm:currentLifeCycleState = 'canceled') ORDER BY dc:created";
-            try (IterableQueryResult results = session.queryAndFetch(query, "NXQL")) {
-                for (Map<String, Serializable> result : results) {
-                    routeIds.add(result.get("ecm:uuid").toString());
-                    i++;
-                    // stop when the limit is reached and close the resultSet
-                    if (i == limit) {
-                        break;
-                    }
-                }
-            }
-            for (String routeDocId : routeIds) {
-                final String associatedTaskQuery = String.format(
-                        "SELECT ecm:uuid FROM Document WHERE ecm:mixinType = 'Task' AND nt:processId = '%s'",
-                        routeDocId);
-                try (IterableQueryResult tasks = session.queryAndFetch(associatedTaskQuery, "NXQL")) {
-                    for (Map<String, Serializable> task : tasks) {
-                        final String taskId = task.get("ecm:uuid").toString();
-                        session.removeDocument(new IdRef(taskId));
-                    }
-                }
+            PartialList<Map<String, Serializable>> workflows = session.queryProjection(WORKFLOWS_QUERY, limit, 0);
+            numberOfCleanedUpWorkflows = workflows.list.size();
+
+            for (Map<String, Serializable> workflow : workflows) {
+                String routeDocId = workflow.get(ECM_UUID).toString();
+                final String associatedTaskQuery = String.format(TASKS_QUERY, routeDocId);
+                session.queryProjection(associatedTaskQuery, 0, 0)
+                       .stream()
+                       .map((task) -> new IdRef(task.get(ECM_UUID).toString()))
```

Comment:
Probably unneeded parentheses for `(task)`.

Reply:
Is that the official Nuxeo code style? No parentheses on one-liner?
If yes, could it be enforced by the Eclipse formatter that lives in tools/nuxeo_formatter.xml?

Reply:
I mean `map(task -> new IdRef(task.get(ECM_UUID).toString()))`.

It's my understanding that this is the official Java code style, used for instance in https://docs.oracle.com/javase/tutorial/java/javaOO/lambdaexpressions.html so we have no reason to use unneeded parentheses.

I doubt that the Eclipse code style can enforce that...

Reply:
Ok, too bad for Eclipse :(